app/controllers/admin/config/flags_controller.rb

```diff
@@ -34,4 +69,18 @@ def reorder
       end
     end
   end
+
+  def destroy
+    with_service(Flags::DestroyFlag) do
+      on_success do
+        Discourse.request_refresh!
+        render(json: success_json)
+      end
+      on_failure { render(json: failed_json, status: 422) }
+      on_failed_policy(:invalid_access) { raise Discourse::InvalidAccess }
```

Comment:
I think it could be worth adding a separate policy for checking if the flag is used already, otherwise you get this unfriendly message when trying to delete a flag from the admin list that has been used:

![image](https://github.com/discourse/discourse/assets/920448/335094f5-cd11-4678-bd25-70958d7ebda4)

Then we could say specifically "You cannot delete this flag because it has already been used in the review system, however you can still disable it."

Reply:
Also maybe we should disable the Delete button if the flag is used? And show a title on hover with the same message as above


![image](https://github.com/discourse/discourse/assets/920448/becbf364-ddea-41a3-9fa6-ff925230e81e)

Reply:
I am surprised that you still have Edit and Delete buttons visible. They should be gone in the current implementation when the flag was used.

But yeah, agree that the best would be to disable them and inform admins why they are disabled.